devenv

```diff
@@ -2,6 +2,60 @@
 
 PATH="./build:${PATH}"
 
+# MacOS specific checks
+if [ "$(uname)" == "Darwin" ] ; then
+
+  echo "Setting PATH with MacOS specific locations"
+  export PATH="/usr/local/opt/coreutils/libexec/gnubin:$PATH"
+  export PATH="/usr/local/opt/gnu-sed/libexec/gnubin:$PATH"
+  export PATH="/usr/local/opt/make/libexec/gnubin:$PATH"
+
+  if ! grep --version 2>/dev/null | grep -q "GNU grep" ; then
+    cat >&2 <<EOF
+== GNU grep is required. Fix with ==
+
+  brew install grep --with-default-names
+
+EOF
+  fi
+
+  if ! sed --version 2>/dev/null | grep -q "GNU sed" ; then
+    cat >&2 <<EOF
+== GNU sed is required. Fix with ==
+
+  brew install gnu-sed
+
+EOF
+  fi
+
+  if ! readlink --version 2>/dev/null | grep -q "GNU coreutils" ; then
+    cat >&2 <<EOF
+== GNU coreutils is required. Fix with ==
+
+  brew install coreutils  --with-default-names
+
+EOF
+  fi
+
+  # have to use || here because BSD find doesnt even support --version (it errors)
+  # shellcheck disable=SC2185
+  find --version 1>/dev/null 2>/dev/null || cat >&2 <<EOF
+== GNU findutils is required. Fix with ==
+
+    brew install findutils  --with-default-names
+
+EOF
+
+  if grep "osxkeychain" ~/.docker/config.json ; then
+    cat >&2 <<EOF
+== Docker configured with keychain ==
+
+  Remove osxkeychain from ~/.docker/config.json
```

Comment:
I don't think you need this unless you are planning on having docker interact with private repositories in automation and we don't do that in this repository AFAIK

Reply:
ack